### DISCUSSION TITLE
Ariston/ATAG nuos HP Boiler Bus Bridgenet addendum

### DISCUSSION THREAD

**Discussion by SdrEngineering** [wrongisthenewright/ebusd-configuration-ariston-bridgenet#6] [General]
I have a ATAG/Ariston nuos heatpump boiler which I am trying to interface with ebusd. Now I noticed with the documentation they also provided an addendum for the bus bridgenet which is not available online. Maybe this helps to expand the configuration file also for the Nuos line of heat pump boilers. 

![photo1679085467 (1)](https://user-images.githubusercontent.com/77666590/226037971-18d769e4-7016-4809-8fde-dba8d152bab6.jpeg)
![photo1679085467](https://user-images.githubusercontent.com/77666590/226038070-8b0c3544-f73a-4d0e-99b1-0a55592ef249.jpeg)

**Comment by wrongisthenewright** [comment]
Hi, unluckily, as far as I know, there's no correlation between Bridgenet command codes and remote controller menu items. I can't help much having a totaly different system, With only the handbook/manual is impossible to try a decoding. 
Have you applied my config file to your appliance? Whic values are working?

If you can you should start grabbing the messages travelling on the bus (ebusd in direct mode) while using the remote controller to scroll in the menus.
With a lot of patience you should be able to map these raw messages to the actual menu items. ebusctl grab all decode comand can help in finding the links betrwwn value in messages with values read by the remote controller.

Then the decoding is usually simple: almost all temp values are SIN data, all stauses are BCD.

Best of luck!

**Reply by SdrEngineering** [reply]
If there is no correlation between the command codes and the menu items, then the remote should contain the command mapping of all devices (including future devices) or the software of the remote should be updateable. I don't know if this is the case for the Ariston sensys for example which should be compatible to this boiler (not sold around here unfortunately). Otherwise I would expect the command code to be some function of the menu item number.

I'm still fighting a bit with ebusd to configure it to use the ariston.csv file and how to use/interpret it (not a software engineer as well). I would expect at least part of the messages to be covered by your configuration file, some of the command codes appear in the messages being transmitted from the boiler.

**Reply by wrongisthenewright** [reply]
Sorry but I may have mislead you. What I was saying is that from the remote command code (sensys in my case) eg. 33.0.0 there's no way to detect the correct ebus command code. If tou look at my CSV you'll notice that the ebus codes are couple of almost random hex codes. I'm surely not able to geuss the ebis command codes simply having the list of remote command menu entry.

I think that your series of appliance use a different range of codes from the one I'm seeing, my Sensys doesn't have a 33.x.x menu item, the device stops at 20.x.x or something like that.

For ebusd and the use of CSV take a look at ebusd wiki page, it's complicated but with a little time you should bi able tofigure put what's needed.

Let me know if I can help.
